src/main/java/Duke.java

```diff
@@ -1,10 +1,138 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
     public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> taskList = new ArrayList<>();
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        greet();
+        String username;
+        username = in.nextLine().trim();
+        System.out.println("Hello " + username + "! What can I do for you?");
+        addHorizontalLine();
+        while (true) {
+            String userRequest = in.nextLine();
+            if (userRequest.equals("list")) {
+                int listIndex = 1;
+                addHorizontalLine();
+                System.out.println("Here are the tasks in your list:");
+                for (Task task : taskList) {
+                    System.out.println(listIndex + ". " + task.toString());
+                    listIndex++;
+                }
+                addHorizontalLine();
+            } else if (userRequest.equals("bye")) {
+                bye();
+                return;
+            } else if (userRequest.startsWith("done") && (userRequest.split(" ").length == 2)) {
+                done(taskList, userRequest);
+            } else {
+                TaskType newRequest = extractTaskType(userRequest);
+                Task taskEntry = null;
+                if (newRequest != null) {
+                    switch (newRequest) {
+                    case DEADLINE:
+                        String deadlineName = getTaskName(userRequest);
+                        String deadlineDetail = getTaskDetail(userRequest);
+                        taskEntry = new Deadline(deadlineName, deadlineDetail);
+                        break;
+
+                    case EVENT:
+                        String eventName = getTaskName(userRequest);
+                        String eventDetail = getTaskDetail(userRequest);
+                        taskEntry = new Event(eventName, eventDetail);
+                        break;
+
+                    case TODO:
+                        String toDoName = getToDoName(userRequest);
+                        taskEntry = new ToDo(toDoName);
+                        break;
+                    default:
```

Comment:
For consistency, should there be an empty line here (since you left empty lines between the other cases)?